lua/entities/ent_jack_gmod_ezcsparticle.lua

```diff
@@ -0,0 +1,216 @@
+-- Based off of JMOD EZ Gas Particle, created by Freaking Fission, uses some code from GChem
+AddCSLuaFile()
+ENT.Type="anim"
+ENT.PrintName="EZ CS Gas"
+ENT.Author="Jackarunda, Freaking Fission"
+ENT.NoSitAllowed=true
+ENT.Editable=true
+ENT.Spawnable=false
+ENT.AdminSpawnable=false
+ENT.AdminOnly=false
+ENT.RenderGroup=RENDERGROUP_TRANSLUCENT
+ENT.EZgasParticle=true
+
+function ENT:SetupDataTables()
+	self:NetworkVar("Int",0,"Siz")
+	self:NetworkVar("Int",1,"Opacity")
+end
+
+if(SERVER)then
+	function ENT:Initialize()
+		local Time=CurTime()
+		self.LifeTime=math.random(50,100)*JMOD_CONFIG.PoisonGasLingerTime
+		self.DieTime=Time+self.LifeTime
+		self:SetModel("models/dav0r/hoverball.mdl")
+		self:SetMaterial("models/debug/debugwhite")
+		self:RebuildPhysics()
+		self:DrawShadow(false)
+		self.NextDmg=Time+2.5
+		self:NextThink(Time+.5)
+		self:SetSiz(1)
+		self:SetOpacity(93.75)
+	end
+	function ENT:ShouldDamage(ent)
+		if not(IsValid(ent))then return end
+		if(ent:IsPlayer())then return ent:Alive() end
+		if((ent:IsNPC())and(ent.Health)and(ent:Health()))then
+			local Phys=ent:GetPhysicsObject()
+			if(IsValid(Phys))then
+				local Mat=Phys:GetMaterial()
+				if(Mat)then
+					if(Mat=="metal")then return false end
+					if(Mat=="default")then return false end
+				end
+			end
+			return ent:Health()>0
+		end
+		return false
+	end
+	function ENT:CanSee(ent)
+		local Tr=util.TraceLine({
+			start=self:GetPos(),
+			endpos=ent:GetPos(),
+			filter={self,ent},
+			mask=MASK_SHOT
+		})
+		return not Tr.Hit
+	end
+	function ENT:Think()
+		if(CLIENT)then return end
+		local Time,SelfPos=CurTime(),self:GetPos()
+		if(self.DieTime<Time)then self:Remove() return end
+		local Force=VectorRand()*10
+		for key,obj in pairs(ents.FindInSphere(SelfPos,300))do
+			if(not(obj==self)and(self:CanSee(obj)))then
+				local distanceBetween = SelfPos:DistToSqr (obj:GetPos())
+				if(not obj.EZgasParticle)then
+					if((self:ShouldDamage(obj)) and self.NextDmg<Time)then
+						
+						FaceProtected = false
```

Comment:
make this local
and the next one too, DEFINITELY local

Reply:
make faceprotected local, but what else?

Reply:
yes, make them both local... remember that a bunch of these particles are operating together, so we don't want them interfering with eachother